src/Extensions/Diagnostics.HealthChecks/AbstractHealthCheck.cs

```diff
@@ -49,6 +49,7 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
 			try
 			{
 				HealthCheckResult result = await CheckHealthInternalAsync(context, token).ConfigureAwait(false);
+				result = EnforceFailureStatus(context.Registration.FailureStatus, result);
```

Comment:
Is it worth having a new function for this that is only used here instead of using the else branch in that function directly here? Are there other places where this function will potentially be used?

Reply:
It's a form of self-documentation. If a piece of code deserves a comment, it probably deserves a name, and here I certainly want to explain what's happening.